src/main/java/org/generationcp/middleware/operation/parser/WorkbookParser.java

```diff
@@ -443,105 +421,88 @@ protected Date validateDate(final String dateString, final boolean isStartDate,
-			this.validateLabelBasedOnSection(sectionName, measurementVariableFromParsedData, displayRowNumber);
-			this.assignRoleBasedOnSectionName(sectionName, measurementVariableFromParsedData, displayRowNumber);
-			this.assignVariableType(sectionName, measurementVariableFromParsedData, this.currentWorkbook);
-
-			measurementVariables.add(measurementVariableFromParsedData);
-
-			this.currentRowZeroBased++;
-		}
-	}
-
-	protected void assignVariableType(final String name, final MeasurementVariable measurementVariable,
-			final org.generationcp.middleware.domain.etl.Workbook workbook) {
-
-		// NOTE: Explicitly setting variable type
-		if (Section.CONSTANT.toString().equals(name) && workbook != null) {
-			final StudyTypeDto studyType = workbook.getStudyDetails().getStudyType();
-			measurementVariable.setVariableType(VariableType.STUDY_CONDITION);
-		} else {
+			final MeasurementVariable measurementVariable = new MeasurementVariable();
 			measurementVariable
-					.setVariableType(OntologyDataHelper.mapFromPhenotype(measurementVariable.getRole(), measurementVariable.getProperty()));
-		}
-
-	}
-
-	protected void assignRoleBasedOnSectionName(final String sectionName, final MeasurementVariable measurementVariable,
-			final Integer rowNumber) {
-
-		if (Section.VARIATE.toString().equals(sectionName) || Section.CONSTANT.toString().equals(sectionName)) {
+				.setName(WorkbookParser.getCellStringValue(workbook, WorkbookParser.DESCRIPTION_SHEET, this.rowIndex, 0));
+			measurementVariable
+				.setDescription(WorkbookParser.getCellStringValue(workbook, WorkbookParser.DESCRIPTION_SHEET, this.rowIndex, 1));
+			measurementVariable.setCropOntology(
+				WorkbookParser.getCellStringValue(workbook, WorkbookParser.DESCRIPTION_SHEET, this.rowIndex, 2));
+			measurementVariable
+				.setProperty(WorkbookParser.getCellStringValue(workbook, WorkbookParser.DESCRIPTION_SHEET, this.rowIndex, 3));
+			measurementVariable
+				.setScale(WorkbookParser.getCellStringValue(workbook, WorkbookParser.DESCRIPTION_SHEET, this.rowIndex, 4));
+			measurementVariable
+				.setMethod(WorkbookParser.getCellStringValue(workbook, WorkbookParser.DESCRIPTION_SHEET, this.rowIndex, 5));
+			measurementVariable
+				.setDataType(WorkbookParser.getCellStringValue(workbook, WorkbookParser.DESCRIPTION_SHEET, this.rowIndex, 6));
+			measurementVariable
+				.setValue(WorkbookParser.getCellStringValue(workbook, WorkbookParser.DESCRIPTION_SHEET, this.rowIndex, 7));
+
+			this.validateRequiredFields(measurementVariable, displayRowNumber);
+			this.validateDataTypeIfNecessary(measurementVariable, displayRowNumber);
+			this.assignVariableTypeAndRoleBasedOnSectionName(sectionName, measurementVariable);
+
+			measurementVariables.add(measurementVariable);
+
+			this.rowIndex++;
+		}
+	}
+
+	protected void assignVariableTypeAndRoleBasedOnSectionName(final String sectionName, final MeasurementVariable measurementVariable) {
+		if (Section.STUDY_DETAILS.getName().equals(sectionName)) {
```

Comment:
I would role, label and VariableType as properties of Section enum to avoid this series of if-else statements